toolkit/tools/internal/packagerepo/repocloner/rpmrepocloner/rpmrepocloner.go

```diff
@@ -616,7 +617,7 @@ func (r *RpmRepoCloner) clonePackage(baseArgs []string) (preBuilt bool, err erro
 		finalArgs := append(baseArgs, reposArgs...)
 
 		// We run in a retry loop on errors deemed retriable.
-		cancel := make(chan struct{})
+		ctx, cancelFunc := context.WithCancel(context.Background())
```

Comment:
done

Reply:
I think it's fine (and expected) to not cancel a context. If I understand this mechanism correctly, we only want to cancel when something special happens. In this case it's an unrecoverable error, which allows us to exit early, since retries won't help.

Reply:
Per the examples in https://go-language.org/go-docs/context/#example_WithCancel the suggested pattern seems to be to use defer. 

A context should be closed when the work is done, not just in the case of error. I agree that "cancel" is an odd name for it, but that does seem to be the expected flow.